### PR TITLE
added closed and destroyed properties on agent

### LINKED ISSUE #607: `Agent` is missing closed and destroyed properties
`Agent` should provide the `closed` and `destroyed` properties similar to `Client`.

### PR DESCRIPTION
**Pull request by yashshah1**
(Tries to) fix #607.

Please let me know if I've missed something

### CODE REVIEW

**Comment by ronag** [comment]
Missing tests. Please also take a look how it's implement in `Pool`.

File: lib/agent/index.js [hunk 1]
```diff
@@ -8,11 +8,16 @@ const util = require('../core/util')
...
-const kFactory = Symbol('factory')
+const {
+  kOnConnect,
+  kOnDisconnect,
+  kClients,
+  kMaxRedirections,
+  kFactory,
+  kClosed,
+  kDestroyed
+} = require("../core/symbols")
```
Comment (ronag) [inline lib/agent/index.js:19]: Unrelated change. You can just put the symbols in this file like the rest.

File: lib/agent/index.js [hunk 2]
```diff
@@ -122,15 +137,25 @@ class Agent extends EventEmitter {
     for (const pool of this[kClients].values()) {
       closePromises.push(pool.close())
     }
-    return Promise.all(closePromises)
+    return Promise.all(closePromises).then((closedPromises) => {
+      return new Promise((resolve) => {
+        this[kClosed] = true
```
Comment (ronag) [inline lib/agent/index.js:142]: This should be set directly, please see how it's done in `Pool` and `Client`.

File: lib/agent/index.js [hunk 3]
```diff
@@ -122,15 +137,25 @@ class Agent extends EventEmitter {
...
 
   destroy () {
     const destroyPromises = []
     for (const pool of this[kClients].values()) {
       destroyPromises.push(pool.destroy())
     }
-    return Promise.all(destroyPromises)
+    return Promise.all(destroyPromises).then((destroyedPromises) => {
+      return new Promise((resolve) => {
+        this[kDestroyed] = true
```
Comment (ronag) [inline lib/agent/index.js:155]: This should be set directly, please see how it's done in `Pool` and `Client`. Also should set `this[kClosed]`.

**Review by ronag — CHANGES_REQUESTED** [review]
Please fix comments and conflicts.

**Comment by ronag** [comment]
ping @yashshah1

**Comment by yashshah1** [comment]
Hi, sorry was busy with a little work and a few family things, pushing all my changes today. Thank you for your patience

**Comment by yashshah1** [comment]
The file `lib/agent.js` already has implementations for `closed` and `destroyed` which were added in 9c04cfa34999. It was missing a getter for both, which I've added. Am I missing something?

**Comment by mcollina** [comment]
CI is failing and it's missing tests

**Comment by ronag** [comment]
I'll add tests later.

**Pull request merged**